### ISSUE TITLE
[DOCS] Create a tutorial about using SpanMarker and Argilla

### ISSUE THREAD

**Issue by sdiazlor** [recognai/rubrix#4086]
## Which page or section is this issue related to?

To create a tutorial about how to use SpanMaker and Argilla for NER

**Comment by davidberenstein1957** [comment]
https://huggingface.co/datasets?task_ids=task_ids%3Anamed-entity-recognition

**Comment by davidberenstein1957** [comment]
https://docs.argilla.io/en/latest/tutorials/notebooks/ner_fine_tune_bert_beginners.html

**Comment by davidberenstein1957** [comment]
https://www.numind.ai/blog/a-foundation-model-for-entity-recognition

**Comment by davidberenstein1957** [comment]
https://huggingface.co/numind/generic-entity_recognition_NER-v1

**Comment by Rami-Ismael** [comment]
Hi there! I'm Rami Ismael, the individual behind the GitHub issues initiative [as discussed here](https://rubrixworkspace.slack.com/archives/C05BCGM277B/p1698662698127679). I'm currently enjoying my winter break and have some free time on my hands. I'm keen on offering my assistance to help finalize the documentation. Would that be possible?

**Comment by nataliaElv** [comment]
Perhaps this tutorial will be more useful when we release the Spans Question for Feedback Datasets? Otherwise it will be outdated quite soon.

**Comment by ceteri** [comment]
We're building this <https://github.com/DerwenAI/textgraphs> which leverages `SpanMarker` and other LLM-based tasks in KG construction ... and if you notice the "report" this project has a very large `Argilla`-shaped puzzle piece missing in its center (why we needed the gradients for extracted entity and relation streams). I'd like to offer help on the `SpanMarker` + `Argilla` tutorial too.

**Comment by louisguitton** [comment]
I'd also like to offer help on this tutorial, whether on designing it, writing it or maintaining it.

My notes on writing a tutorial:
- the end of the tutorial must be meaningful and achievable to a beginner
- having done the tutorial, the reader is in position to make sense of the rest of the documentation and of Argilla itself
- objective = turning learners into users, get the learner started on their Argilla journey not to their destination
- Tutorials need to be useful for the beginner, easy to follow, meaningful and extremely robust, and kept up-to-date
- build from the simplest tools or operations to the most complex
- be concrete, built with specificity in mind, don't explain anything the learner doesn't need to know to complete the tutorial (e.g. Argilla telemetry)
- Note that it doesn’t tell you what you will learn, just what you will do. The learning comes out of that doing.

Proposed promise for this Argilla + SpanMarker tutorial
> if you have the basic knowledge required to follow this tutorial (e.g. spaCy?), and you follow its directions, you will end up with a working _Argilla Server_, complete with a _FeedbackDataset_ with Span Categorization _Questions_, with NER label _Suggestions_ machine-generated by SpanMarker, ready for _Annotators_ to add _Responses_. Advanced readers will be able to add _Metadata_ or _Vectors_.

What do you think? what would be a good amount of knowledge required?

And we are waiting for the Span Categorization to be released in the FeedbackDataset right? or did I miss this going live?

**Comment by nataliaElv** [comment]
hi @louisguitton ! Yes, we're working on releasing a Spans question for Feedback datasets and once that's out, we can start working on the tutorial. I think it would be highly beneficial for the adoption of this feature that the tutorial is published soon after the release.

I'll leave it to @davidberenstein1957 and @sdiazlor to tell you if they need any help with this one or if it's something they prefer to do internally.

**Comment by dvsrepo** [comment]
Very cool notes about what a tutorial should be @louisguitton , fully agree!

We used to use tutorials more as a blog post to promote and introduce argilla to new users on social media but that has created a bit of a mismatch now. We'll take it into account for version 2.0 of the docs (unstarted but planned)

**Comment by sdiazlor** [comment]
Hi, @louisguitton. Thanks for your notes! Any feedback is always welcome. Feel free to work on this tutorial and let us know if you have any doubt.

**Comment by dvsrepo** [comment]
Let's wait for the new SpanQuestion

**Comment by louisguitton** [comment]
Since we started this discussion, SpanQuestion was released
- V1.26 Mar 22, 2024 SpanQuestion is now part of FeedbackDataset #4617, #4623, #4622
- V1.27 Apr 18, 2024 Overlapping spans are now possible #4668, #4697
- V1.28 May 9, 2024 Span improvements #4735, #4726
- [NER on Argilla - meetup talk - Louis Guitton](https://www.youtube.com/watch?v=VdW-cqD4w9I) with code available here (not cleaned up) https://github.com/louisguitton/mlops-talk-llm-kg/tree/main/notebooks/argilla_talk

The football news dataset, the code snippets I contribute in the talk, the structure of the talk can all be used to create a tutorial.
A part 2 of the talk was also discussed, to address some of the parts I didn't have time to cover: train a model, use weak supervision with skweak, do KG construction with the entities found etc...

The scoping exercise (i.e. splitting in parts and making sure we deliver small and incremental value) for NER is key I think, so any input from User feedback or Customer needs or Product vision is welcome to help prioritise.
